src/sensors.py

```diff
@@ -75,28 +82,47 @@ def get_updates():
 
 # Temperature method depending on system distro
 def get_temp():
-    temp = ''
-    if 'rasp' in OS_DATA['ID']:
-        reading = subprocess.check_output(['vcgencmd', 'measure_temp']).decode('UTF-8')
-        temp = str(re.findall('\d+.\d+', reading)[0])
-    else:
-        reading = subprocess.check_output(['cat', '/sys/class/thermal/thermal_zone0/temp']).decode('UTF-8')
-        temp = str(reading[0] + reading[1] + '.' + reading[2]) # why?? need linux system to test
+    temp = 'Unknown'
+    # Utilising psutil for temp reading on ARM arch
+    try:
+        temp = psutil.sensors_temperatures()['cpu_thermal'][0].current
+    except:
+        try:
+            # Assumes that first entry is the CPU package, have not tested this on other systems except my NUC x86
+            temp = psutil.sensors_temperatures()['coretemp'][0].current
+        except Exception as e:
+            print('Could not establish CPU temperature reading: ' + str(e))
+            raise
```

Comment:
I was debating doing away with a seperate call for Raspi, but the comments over in [this SO thread](https://raspberrypi.stackexchange.com/questions/41784/temperature-differences-between-cpu-gpu) suggest that `thermal_zone` is an alias of `vcgencmd` on the Pi and that the latter is the most direct way of getting the data. I still don't think the performance increase will be anywhere near perceivable, so using a single command would be better and far more maintainable.
While I can't verify this, [this issue](https://github.com/j7126/OctoPrint-Dashboard/issues/148) reports an as-of-yet unfixed bug with `psutil` temperature readings when `1wire` sensors are connected via gpio. I would be in favour of getting the required data in the most direct way possible to avoid errors caused by dependencies; at least wherever it's not a hassle to reinvent the wheel.

Reply:
Thanks a ton for reviewing the PR! I appreciate your insight and for weighing in. It's definitely a good call to keep things simple.

The primary reason for the change, was because with the previous method's `thermal_zone`, users would have to identify which thermal zone their CPU is on, and then add that to an additional CPU Zone config yaml parameter. I felt like that was too many steps, so opted for a more streamlined solution and just put it all in one common basket - but you bring up great points.

### Raspberry PI / ARM

On systems that only list a single `thermal_zone`, like Rasperry PIs systems, it totally makes sense to use that direct method. I punched out some quick comparisons between `thermal_zone` and `vcgencmd`, and they both jam out very similar values. Assuming the first link you sent correctly assumed they're they same thing.

```
pi@PiZero32:~/system_sensors/src $ vcgencmd measure_temp
temp=40.6'C
pi@PiZero32:~/system_sensors/src $ sudo cat /sys/class/thermal/thermal_zone0/temp
40084
pi@PiZero32:~/system_sensors/src $ vcgencmd measure_temp
temp=39.0'C
pi@PiZero32:~/system_sensors/src $ sudo cat /sys/class/thermal/thermal_zone0/temp
39546
pi@PiZero32:~/system_sensors/src $ vcgencmd measure_temp
temp=40.1'C
pi@PiZero32:~/system_sensors/src $ sudo cat /sys/class/thermal/thermal_zone0/temp
39546
pi@PiZero32:~/system_sensors/src $ 
```

### Potential `psutil` / 1-wire bug

As for the concern about `psutil`'s potential conflict while using 1-wire, I fortunately have a 1-wire temperature sensor running on a Pi Zero that tracks my server rack temperature. The same Pi Zero is running this `System Sensors` PR (that uses only `psutil` for the CPU temperature), and displays both the 1-wire and CPU temperatures correctly.

Of course, that's a test with one system, but unless anyone else bumps into this issue, I feel like using `psutil` for non-ARM systems will make it far easier than having users fiddling around to find the correct `thermal_zone` for their CPU.

![1-wire_psutil](https://user-images.githubusercontent.com/3183008/133061166-91d909eb-9115-40b4-b578-aca2b26e1827.jpg)



So... how about something like:

```
try:
    vcgencmd measure_temp
exception:
   psutil.....
```

Thoughts @SvbZ3r0 ?

Reply:
Sounds good!
In fact, considering that vcgencmd works only on Raspberry Pis and that the Pi has only one temp sensor (so vcgencmd, thermal_zone and hwmon/temp (where psutil gets its data from) are all the same thing anyway), if psutil doesn't have a bug, we could just stick with psutil for all architectures.
We'll have to wait for @Sennevds to weigh in when he has the time.